src/hclib-promise.c

```diff
@@ -110,10 +100,8 @@ hclib_promise_t **hclib_promise_create_n(size_t nb_promises,
  * Note: this is concurrent with the 'put' operation.
  */
 void *hclib_future_get(hclib_future_t *future) {
-    if (future->owner->datum == UNINITIALIZED_PROMISE_DATA_PTR) {
-        return NULL;
-    }
-    return (void *)future->owner->datum;
```

Comment:
This is related to the `volatile void *datum` change. This cast (and the similar casts in `hclib_future_wait` are necessary because `datum` is a pointer to *volatile data*. After correcting the declaration, these casts are no longer necessary. The fact that we always discarded the `volatile` qualifier before dereferencing the `datum` pointer was a strong indicator that the declaration was incorrect.